sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorEventSource.cs

```diff
@@ -0,0 +1,452 @@
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(14, Level = EventLevel.Error, Message = "Failed to renew ownership. (Identifier: '{0}'; ErrorMessage: '{0}')")]
+        public virtual void RenewOwnershipError(string identifier, string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(14, identifier ?? string.Empty, errorMessage ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an attempt to renew ownership has completed, so they don't expire.
+        /// </summary>
+        ///
+        /// <param name="identifier">A unique name used to identify the event processor.</param>
+        ///
+        [Event(15, Level = EventLevel.Verbose, Message = "Attempt to renew ownership has completed. (Identifier: '{0}')")]
+        public virtual void RenewOwnershipComplete(string identifier)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(15, identifier ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Creates and starts running a new partition processing task.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is starting.</param>
+        ///
+        [Event(16, Level = EventLevel.Informational, Message = "Creating and start running a new partition processing task for partition id '{0}'.")]
+        public virtual void StartPartitionProcessing(string partitionId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(16, partitionId ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while running a new partition processing task.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(17, Level = EventLevel.Error, Message = "Failed to Create and start running a new partition processing task for partition '{0}'. (ErrorMessage: '{1}')")]
+        public virtual void StartPartitionProcessingError(string partitionId,
+                                                          string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(17, partitionId ?? string.Empty, errorMessage ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Partition processing task has completed.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is starting.</param>
+        ///
+        [Event(18, Level = EventLevel.Verbose, Message = "Partition processing task for partition id '{0}' has completed.")]
```

Comment:
I believe we can not decide `EventLevel` by checking that this doesn't happen very often so we should set this as `Verbose` or this happens very often so we should set this as `Informational`.  I think It depends on what level of information we should provide users for the particular call.

Reply:
I would definitely say that we should pair the start/stop for each in the same event level.   Errors in that operation, obviously, have the appropriate already.   It would be confusing to see a bunch of logs for "starting XXX" and not see them book-ended with the "Done with XXX"

Reply:
Set the same event level for start/stop - https://github.com/Azure/azure-sdk-for-net/pull/9310